waltid-services/waltid-wallet-api/Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM gradle:latest AS pre-cache
+COPY . /home/src
+COPY gradle.properties settings.gradle.kts /home/app/
+COPY gradle /home/app/gradle
+RUN cd /home/src && \
+    find . -name "build.gradle.kts" -exec sh -c 'mkdir -p /home/app/$(dirname "$1"); cp "$1" /home/app/$1' _ {} \;
+
+# Stage 1: Cache Gradle dependencies
+FROM gradle:latest AS cache
+RUN mkdir -p /home/gradle/cache_home
+ENV GRADLE_USER_HOME=/home/gradle/cache_home
+COPY --from=pre-cache /home/app/ /home/gradle/app/
+WORKDIR /home/gradle/app/waltid-services/waltid-wallet-api
+RUN gradle dependencies --no-daemon
+
+# Stage 2: Build Application
+FROM gradle:latest AS build
+COPY --from=cache /home/gradle/cache_home /home/gradle/.gradle
+COPY --chown=gradle:gradle . /home/gradle/src
+WORKDIR /home/gradle/src/waltid-services/waltid-wallet-api
+# Build the fat JAR, Gradle also supports shadow
+# and boot JAR by default.
+RUN gradle buildFatJar --no-daemon
+
+# Stage 3: Create the Runtime Image
+FROM amazoncorretto:22 AS runtime
```

Comment:
this time wallet-api targets [jvm21](https://github.com/walt-id/waltid-identity/blob/8fcefa7d4f599104fbcffc982133848eef3035ac/waltid-services/waltid-wallet-api/build.gradle.kts#L31)
but I also don't think it would be a problem